src/pages/AddServers.js

```diff
@@ -150,11 +151,20 @@ class AddServers extends BaseUpdateWizardPage {
   // values, then set size is the same as the array list length,
   // otherwise the list contains duplicate values.
   hasDuplicates = (arrayList) => {
-    return (new Set(arrayList)).size !== arrayList.length;
+    if(isEmpty(arrayList)) {
+      return false;
+    }
+    // filter out empty items
+    let cleanList = arrayList.filter(item => !isEmpty(item));
+    if(isEmpty(cleanList)) {
+      return false;
+    }
+
+    return (new Set(cleanList)).size !== cleanList.length;
   }
 
-  hasAddressesConflicts = () => {
-    let hasConflicts = false;
```

Comment:
instead of having this variable `hasInvalid`, the logic would be simpler if you just immediately return when you spot a violation. Also, it would be simpler to understand if this function were in the positive, i.e. `isServerValid`, and you return `true` when it is valid and `false` otherwise.

Reply:
When functions are coded to check whether something is valid or invalid, it is generally easier to understand them if true represent the valid case and false represents invalid.

Reply:
updated with valid is true invalid is false now.